modules/search/reddit.py

```diff
@@ -0,0 +1,101 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from core.module import BaseModule
+import re
+import requests
+
+
+class Module(BaseModule):
+    meta = {
+        'name': 'Reddit Search',
+        'author': 'Kunal Khandelwal',
+        'version': '0.1',
+        'description': 'Search your query in the Reddit and show the results.',
+        'sources': ('google', 'yahoo', 'bing  '),
+        'options': (
+            ('query', None, True, 'Query string', '-q', 'store'),
+            ('limit', 1, False, 'Search limit(number of pages, default=1)', '-l', 'store'),
+            ('count', 50, False, 'Number of links per page(min=10, max=100, default=50)', '-c', 'store'),
+            ('engine', 'google', False, 'Engine names for search(default=google)', '-e', 'store'),
+            ('output', False, False, 'Save output to workspace', '--output', 'store_true'),
+        ),
+        'examples': ('reddit -q <QUERY> -l 15 --output',)
+    }
+
+    def module_run(self):
+        query = self.options['query']
+        limit = self.options['limit']
+        count = self.options['count']
+        engine = self.options['engine'].split(',')
+        google_q = f"site:www.reddit.com inurl:{query}"
+        q = f"site:www.reddit.com {query}"
+        usernames = []
+
+        run = self.google(q, limit, count)
+        run.run_crawl()
+        links = run.links
+        pages = run.pages
+
+        if 'bing' in engine:
+            run = self.bing(q, limit, count)
+            run.run_crawl()
+            pages += run.pages
+            for item in run.links_with_title:
+                link, title = item
+                links.append(link)
+
+        if 'yahoo' in engine:
+            run = self.yahoo(q, limit, count)
+            run.run_crawl()
+            pages += run.pages
+            links += run.links
+            pages += run.pages
```

Comment:
Why did you save pages. duplicate lines(67,65).
There is no need to save pages

Reply:
Since I was writing a module for the first time, I was trying to follow the pattern as per other modules which includes this line. But now I have made the change. Thanks!

Regards
Kunal